### ISSUE TITLE
-tc does not work with comma in names

### ISSUE THREAD

**Issue by martinus** [doctest/doctest#398]
A simple reproducer is this:

```cpp
TEST_CASE("a,b") {}
```

It doesn't seem to be possible to run this test with `-tc`. I can run `./app -ltc` which prints

```
a,b
```

But running `./app -tc='a,b'` tries to run test `a` and test `b`, which both don't exist.

That's not a big problem in that case because I can just rename the test, but it becomes a problem with `TEST_CASE_TEMPLATE`. E.g. see this reproducer:

```cpp
template <typename X, typename Y>
class Foo {};
TYPE_TO_STRING(Foo<int, int>);

TEST_CASE_TEMPLATE("foo", T, Foo<int, int>) {}
```

Running `./app -ltc` prints 

```
foo<Foo<int, int>>
```

But since the test contains a `,` it is not possible to run just this test :cry:

**Comment by onqtam** [comment]
Sorry for the late reply! This issue is the same for filtering test cases and [subcases](https://github.com/onqtam/doctest/issues/135) - the best I could offer for now is to use `?` (will match 1 character) instead of the commas (`,`), so in your case you could use `a?b`. Maybe I should put this in the FAQ & the relevant places in the docs for now... and also maybe escaping should be possible. Will keep this open until the problem is resolved.

**Comment by onqtam** [comment]
a PR which allows commas to be escaped (with `\` as a prefix) was just merged in `dev`, however, I do think that this solution (as proposed in #135 ):
> or change the argument syntax so that if you want to run multiple subcases, you would type --sc='ab' --sc='cd'

would be better in the long term, so I'll keep this issue open

**Comment by Saalvage** [comment]
Primary issue fixed, #135 will remain as the potential rest of this is figured out.
